src/bag/layout/template.py

```diff
@@ -2691,6 +2691,66 @@ def do_power_fill(self, layer_id: int, tr_manager: TrackManager,
             self.draw_vias_on_intersections(vss_warrs, top_vss)
         return top_vdd, top_vss
 
+    def do_multi_power_fill(self, layer_id: int, tr_manager: TrackManager, sup_list: List[Union[WireArray, List[WireArray]]],
+                            bound_box: Optional[BBox] = None, x_margin: int = 0, y_margin: int = 0,
+                            uniform_grid: bool = False) -> List[List[WireArray]]:
+        """Draw power fill on the given layer. Accepts as many different supply nets as provided.
+
+        Parameters
+        ----------
+        layer_id : int
+            the layer ID on which to draw power fill.
+        tr_manager : TrackManager
+            the TrackManager object.
+        sup_list : List[Union[WireArray, List[WireArray]]]
+            a list of supply wires to draw power fill for.
+        bound_box : Optional[BBox]
+            bound box over which to draw the power fill
+        x_margin : int
+            keepout margin on the x-axis. Fill is centered within margin.
+        y_margin : int
+            keepout margin on the y-axis. Fill is centered within margin.
+        uniform_grid : bool
+            draw power fill on a common grid instead of dense packing.
+
+        Returns
+        -------
+        List[List[WireArray]]
+            List of the wire arrays for each supply in sup_list, given in the
+            same order as sup_list.
+        """
+        if bound_box is None:
+            if self.bound_box is None:
+                raise ValueError("bound_box is not set")
+            bound_box = self.bound_box
+        bound_box = bound_box.expand(dx=-x_margin, dy=-y_margin)
+        is_horizontal = (self.grid.get_direction(layer_id) == 0)
+        if is_horizontal:
+            cl, cu = bound_box.yl, bound_box.yh
+            lower, upper = bound_box.xl, bound_box.xh
+        else:
+            cl, cu = bound_box.xl, bound_box.xh
+            lower, upper = bound_box.yl, bound_box.yh
+        fill_width = tr_manager.get_width(layer_id, 'sup')
+        fill_space = tr_manager.get_sep(layer_id, ('sup', 'sup'))
+        sep_margin = tr_manager.get_sep(layer_id, ('sup', ''))
+        tr_bot = self.grid.coord_to_track(layer_id, cl, mode=RoundMode.GREATER_EQ)
+        tr_top = self.grid.coord_to_track(layer_id, cu, mode=RoundMode.LESS_EQ)
+        trs = self.get_available_tracks(layer_id, tid_lo=tr_bot, tid_hi=tr_top, lower=lower, upper=upper,
+                                        width=fill_width, sep=fill_space, sep_margin=sep_margin,
+                                        uniform_grid=uniform_grid)
+        all_wars = [[] for _ in range(len(sup_list))]
```

Comment:
len(sup_list) is used multiple times in the following lines, so just make a new variable for it instead of querying len() every time. Also rename all_wars to all_warrs for sake of consistency.